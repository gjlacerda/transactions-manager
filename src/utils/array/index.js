const removeDuplicates = (
  listFrom = [],
  listTarget = [],
  indexProp,
) => listTarget.reduce((unique, o) => {
  if (!listFrom.some(obj => obj[indexProp] === o[indexProp])) {
    unique.push(o)
  }
  return unique
}, [])

export {
  removeDuplicates,
}
