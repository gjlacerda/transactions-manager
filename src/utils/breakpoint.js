export const sm = '600px'
