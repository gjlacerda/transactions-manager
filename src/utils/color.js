export const colorApp = '#1976d2'
export const colorBlack = '#555'
export const colorWhite = '#ffffff'
