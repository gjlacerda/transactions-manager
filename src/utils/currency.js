export const formatToHuman = value => new Intl.NumberFormat('de-DE').format(value)

export const formatToMachine = value => +value
  .replace(/\./g, '')
  .replace(',', '.')
