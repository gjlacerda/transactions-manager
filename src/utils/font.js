export const fontSize = {
  small: '12px',
  normal: '16px',
  large: '20px',
}

export const fontWeight = {
  regular: 400,
  bold: 700,
}
