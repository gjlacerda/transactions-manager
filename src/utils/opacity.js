export const opacityLight = '0.8'
