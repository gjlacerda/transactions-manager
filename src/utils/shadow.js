export const boxShadow = '0 2px 3px rgba(0, 0, 0, 0.09)'
export const inputShadow = '0px 1px 5px 0px rgba(0,0,0,0.2), 0px 2px 2px 0px rgba(0,0,0,0.14), 0px 3px 1px -2px rgba(0,0,0,0.12)'
