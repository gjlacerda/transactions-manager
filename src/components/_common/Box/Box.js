import React from 'react'
import styled from 'styled-components'
import { marginSize } from 'utils/margin'

const Container = styled.div`
  margin-bottom: ${props => marginSize[props.marginBottom]};
  margin-left: ${props => marginSize[props.marginLeft]};
  margin-right: ${props => marginSize[props.marginRight]};
  margin-top: ${props => marginSize[props.marginTop]};
`
Container.displayName = 'Container'

export default ({
  marginBottom,
  marginLeft,
  marginRight,
  marginTop,
  children,
}) => (
  <Container
    marginBottom={marginBottom}
    marginLeft={marginLeft}
    marginRight={marginRight}
    marginTop={marginTop}
  >
    {children}
  </Container>
)
