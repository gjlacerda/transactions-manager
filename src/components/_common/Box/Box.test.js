import React from 'react'
import renderer from 'react-test-renderer'
import 'jest-styled-components'
import { marginSize } from 'utils/margin'
import Box from './Box'

describe('Box component', () => {
  it('deve ser igual ao snapshot', () => {
    const tree = renderer.create(
      <Box
        marginTop="small"
        marginRight="medium"
        marginBottom="large"
        marginLeft="small"
      />,
    ).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it(`deve conter o css <margin-top: ${marginSize.small}> ao passar o par창metro (marginTop="small")`, () => {
    const tree = renderer.create(<Box marginTop="small" />).toJSON()
    expect(tree).toHaveStyleRule('margin-top', marginSize.small)
  })

  it(`deve conter o css <margin-right: ${marginSize.medium}> ao passar o par창metro (marginBottom="medium")`, () => {
    const tree = renderer.create(<Box marginRight="medium" />).toJSON()
    expect(tree).toHaveStyleRule('margin-right', marginSize.medium)
  })

  it(`deve conter o css <margin-bottom: ${marginSize.large}> ao passar o par창metro (marginBottom="large")`, () => {
    const tree = renderer.create(<Box marginBottom="large" />).toJSON()
    expect(tree).toHaveStyleRule('margin-bottom', marginSize.large)
  })

  it(`deve conter o css <margin-left: ${marginSize.small}> ao passar o par창metro (marginLeft="small")`, () => {
    const tree = renderer.create(<Box marginLeft="small" />).toJSON()
    expect(tree).toHaveStyleRule('margin-left', marginSize.small)
  })
})
