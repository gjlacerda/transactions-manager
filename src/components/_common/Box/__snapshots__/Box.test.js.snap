// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Box component deve ser igual ao snapshot 1`] = `
.c0 {
  margin-bottom: 40px;
  margin-left: 10px;
  margin-right: 20px;
  margin-top: 10px;
}

<div
  className="c0"
/>
`;
